image_tools/src/showimage.cpp

```diff
@@ -80,73 +86,75 @@ void show_image(
-  rmw_qos_history_policy_t history_policy = rmw_qos_profile_default.history;
-  bool show_camera = true;
-  std::string topic("image");
-
-  // Force flush of the stdout buffer.
-  // This ensures a correct sync of all prints
-  // even when executed simultaneously within a launch file.
-  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
-
-  // Configure demo parameters with command line options.
-  if (!parse_command_options(
-      argc, argv, &depth, &reliability_policy, &history_policy, &show_camera, nullptr, nullptr,
-      nullptr, nullptr, &topic))
-  {
-    return 0;
+  execute();
+}
+
+/// Constructor for setup with command line args
+ShowImage::ShowImage(rclcpp::NodeOptions options, int argc, char ** argv)
+: Node("showimage", options){
+  if(setup(argc, argv)){
+    execute();
+  }else{
+    rclcpp::shutdown();
   }
+}
 
-  if (show_camera) {
+/// Execute main functions with image subscriber and callback
+/**
+ */
+void ShowImage::execute(){
+  if (show_camera_) {
     // Initialize an OpenCV named window called "showimage".
+      // RCLCPP_INFO(this->get_logger(), "SHOWING THE CAMERA");
+
     cv::namedWindow("showimage", cv::WINDOW_AUTOSIZE);
     cv::waitKey(1);
   }
 
-  // Initialize a ROS node.
-  auto node = rclcpp::Node::make_shared("showimage");
-
-  // Set quality of service profile based on command line options.
   auto qos = rclcpp::QoS(
     rclcpp::QoSInitialization(
-      // The history policy determines how messages are saved until taken by
-      // the reader.
-      // KEEP_ALL saves all messages until they are taken.
-      // KEEP_LAST enforces a limit on the number of messages that are saved,
-      // specified by the "depth" parameter.
-      history_policy,
-      // Depth represents how many messages to store in history when the
-      // history policy is KEEP_LAST.
-      depth
+      history_policy_,
+      depth_
   ));
-
-  // The reliability policy can be reliable, meaning that the underlying transport layer will try
-  // ensure that every message gets received in order, or best effort, meaning that the transport
-  // makes no guarantees about the order or reliability of delivery.
-  qos.reliability(reliability_policy);
-
-  auto callback = [show_camera, &node](const sensor_msgs::msg::Image::SharedPtr msg)
+  qos.reliability(reliability_policy_);
+  auto callback = [this](const sensor_msgs::msg::Image::SharedPtr msg)
     {
-      show_image(msg, show_camera, node->get_logger());
+      show_image(msg, show_camera_, this->get_logger());
     };
 
-  std::cerr << "Subscribing to topic '" << topic << "'" << std::endl;
-  RCLCPP_INFO(node->get_logger(), "Subscribing to topic '%s'", topic.c_str());
-  // Initialize a subscriber that will receive the ROS Image message to be displayed.
-  auto sub = node->create_subscription<sensor_msgs::msg::Image>(
-    topic, qos, callback);
+  std::cerr << "Subscribing to topic '" << topic_ << "'" << std::endl;
```

Comment:
I think this can be removed. It looks like debugging leftovers and there is already a log macro with the same info.